File: src/py/pe_0002.py
# FUNCTIONS


def fibonacci(a=1, b=1):
    yield a
    yield from fibonacci(b, a + b)


# SCRIPT


if __name__ == "__main__":
    gross = 0
    for n in fibonacci(1, 2):
        if 4e6 < n:
            break
        elif n % 2 == 0:
            gross += n

    print(gross)

File: src/js/pe_0002.js
var gross = 0;
var a = 1;
var b = 2;

while (a < 4e6) {
    if (a % 2 == 0) {
        gross += a;
    }
    c = a + b;
    a = b;
    b = c;
}

console.log(gross);
